segmentation/utils/visualize.py

```diff
@@ -18,3 +18,84 @@ def visualize_z_scores(z, pheno_titles):
     zplot = pd.DataFrame(z, columns=pheno_titles, index=pheno_titles)
     sns.set(font_scale=.7)
     sns.clustermap(zplot, figsize=(8, 8), cmap="vlag")
+    
+def newDict(names):
+    result = {}
+    for name in names:
+        result[name] = [0]
+    return result
+
+def visualize_distribution_of_cell_count(df, id_col_name, cell_col_name):
+    ids = set(df[id_col_name])
+    print(ids)
+    cell_names = df[cell_col_name].value_counts().index
+    cell_names = cell_names.tolist()  
+
+    unique_ids = []
+    for item in ids:
+        if type(item) is float or type(item) is int:
+            unique_ids.append(item)
+
+    unique_ids = [x for x in unique_ids if x == x]
```

Comment:
I believe a python set automatically only store unique entries.  I don't have too much context for this but aren't id's always floats/ints?  Also, I'm pretty sure line 39 does nothing.  Long story short, I think these lines are unneeded

Reply:
I second Adam on this. 

On a side note, line 39 doesn't work even if you actually had a list with duplicates. That's because you're comparing each element to itself, and such a statement will always be true for integer/float values. You can call set again on unique_ids to properly ensure no duplicates, but in this case, you don't have to cause you pulled items from a set (the id variable) to put into unique_ids.

Reply:
We originally didn't have line 39 but for some reason unique_ids would hold a bunch of NaNs, which is shown below.
![image](https://user-images.githubusercontent.com/32173924/87333430-f1bbca80-c4f1-11ea-93e2-17342770478b.png)
To filter out the NaNs, we test if x==x which doesn't hold true because Nan != Nan

Reply:
Ok, I see, however, the code may be somewhat confusing at first. Aside from putting a comment in to explain what this line does, there are two other possibilities I'd encourage you to look into.

The first is to filter out the NaNs in the if statement. NumPy has a function np.isnan(x) which allows you to check if x is nan or not.

The second is to call np.isnan(x) in your list comprehension.

Reply:
I second Alex's suggestions.  

But I'm curious as to where the NaN's originate.  I'm not sure if it's that meaningful, but if it's from another error earlier in the processing, then we could try to catch it now, and fix it in a later PR.